File: Source/WebDemo/SerenePSM/SerenePSM.Web/Modules/ServerTypes/Default/Pfk7103Form.ts
import { PrefixedContext, initFormType } from "@serenity-is/corelib";
import { StringEditor, DecimalEditor } from "serenity.corelib";

export interface Pfk7103Form {
    PreFix: StringEditor;
    TypeCode: StringEditor;
    SeTypeCode: StringEditor;
    CdTypeCode: StringEditor;
    SeComponentType: StringEditor;
    CdComponentType: StringEditor;
    TypeName: StringEditor;
    TypeNameRelation: StringEditor;
    TypeSimpleName: StringEditor;
    CustomerCode: StringEditor;
    ValueType1: StringEditor;
    QtyBatchS: DecimalEditor;
    QtyDay: DecimalEditor;
    QtyDayS: DecimalEditor;
    QtyMan: DecimalEditor;
    AmtAllowance: DecimalEditor;
    Standard: StringEditor;
    Standard1: StringEditor;
    Standard2: StringEditor;
    Standard3: StringEditor;
    Standard4: StringEditor;
    Standard5: StringEditor;
    DateInsert: StringEditor;
    DateUpdate: StringEditor;
    InchargeInsert: StringEditor;
    InchargeUpdate: StringEditor;
    CheckUse: StringEditor;
    CheckComplete: StringEditor;
    Remark: StringEditor;
    SeSite: StringEditor;
    CdSite: StringEditor;
}

export class Pfk7103Form extends PrefixedContext {
    static readonly formKey = 'Default.Pfk7103';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!Pfk7103Form.init)  {
            Pfk7103Form.init = true;

            var w0 = StringEditor;
            var w1 = DecimalEditor;

            initFormType(Pfk7103Form, [
                'PreFix', w0,
                'TypeCode', w0,
                'SeTypeCode', w0,
                'CdTypeCode', w0,
                'SeComponentType', w0,
                'CdComponentType', w0,
                'TypeName', w0,
                'TypeNameRelation', w0,
                'TypeSimpleName', w0,
                'CustomerCode', w0,
                'ValueType1', w0,
                'QtyBatchS', w1,
                'QtyDay', w1,
                'QtyDayS', w1,
                'QtyMan', w1,
                'AmtAllowance', w1,
                'Standard', w0,
                'Standard1', w0,
                'Standard2', w0,
                'Standard3', w0,
                'Standard4', w0,
                'Standard5', w0,
                'DateInsert', w0,
                'DateUpdate', w0,
                'InchargeInsert', w0,
                'InchargeUpdate', w0,
                'CheckUse', w0,
                'CheckComplete', w0,
                'Remark', w0,
                'SeSite', w0,
                'CdSite', w0
            ]);
        }
    }
}

File: Source/WebDemo/SerenePSM/SerenePSM.Web/Modules/Default/Pfk7103/Pfk7103Form.cs
using Serenity.ComponentModel;

namespace SerenePSM.Default.Forms;

[FormScript("Default.Pfk7103")]
[BasedOnRow(typeof(Pfk7103Row), CheckNames = true)]
public class Pfk7103Form
{
    public string PreFix { get; set; }
    public string TypeCode { get; set; }
    public string SeTypeCode { get; set; }
    public string CdTypeCode { get; set; }
    public string SeComponentType { get; set; }
    public string CdComponentType { get; set; }
    public string TypeName { get; set; }
    public string TypeNameRelation { get; set; }
    public string TypeSimpleName { get; set; }
    public string CustomerCode { get; set; }
    public string ValueType1 { get; set; }
    public decimal QtyBatchS { get; set; }
    public decimal QtyDay { get; set; }
    public decimal QtyDayS { get; set; }
    public decimal QtyMan { get; set; }
    public decimal AmtAllowance { get; set; }
    public string Standard { get; set; }
    public string Standard1 { get; set; }
    public string Standard2 { get; set; }
    public string Standard3 { get; set; }
    public string Standard4 { get; set; }
    public string Standard5 { get; set; }
    public string DateInsert { get; set; }
    public string DateUpdate { get; set; }
    public string InchargeInsert { get; set; }
    public string InchargeUpdate { get; set; }
    public string CheckUse { get; set; }
    public string CheckComplete { get; set; }
    public string Remark { get; set; }
    public string SeSite { get; set; }
    public string CdSite { get; set; }
}